### DISCUSSION TITLE
Evolutionary algorithms

### DISCUSSION THREAD

**Discussion by Koe-eigh** [TimefoldAI/timefold-solver#739] [General]
Are there any ways to implement evolutionary algorithms, such as genetic algorithms, with Timefold? 
If so, how can I implement it?

**Comment by triceo** [comment]
There is no easy way to implement GA in Timefold Solver; practically, you'd need to replace most of the solver with a custom implementation. Many years ago, there were attempts, but we eventually scrapped them when we noticed that Local Search performs significantly better. YMMV.

**Reply by Koe-eigh** [reply]
Thank you for the quick reply!
I'm building a teacher-rostering app and Timefold makes it very efficient! I'm just inquiring if there is another way of making it better! I understand it’s not a simple task, and I’m glad to hear that a local search approach is preferable.”
